pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -94,6 +94,13 @@ type CephClusterList struct {
 }
 
 // ClusterSpec represents the specification of Ceph Cluster
+// +kubebuilder:validation:XValidation:rule="!has(self.external) || (has(self.external) && ((!has(self.external.enable) || self.external.enable == false )|| (self.external.enable == true && (!has(self.mon)))))",message="external mode enabled cluster cannot have mon field in CR"
+// +kubebuilder:validation:XValidation:rule="!has(self.external) || (has(self.external) && ((!has(self.external.enable) || self.external.enable == false )|| (self.external.enable == true && (!has(self.dashboard)))))",message="external mode enabled cluster cannot have dashboard field in CR"
+// +kubebuilder:validation:XValidation:rule="!has(self.external) || (has(self.external) && ((!has(self.external.enable) || self.external.enable == false )|| (self.external.enable == true && (!has(self.monitoring)))))",message="external mode enabled cluster cannot have monitoring field in CR"
+// +kubebuilder:validation:XValidation:rule="!has(self.external) || (has(self.external) && ((!has(self.external.enable) || self.external.enable == false )|| (self.external.enable == true && (!has(self.network)))))",message="external mode enabled cluster cannot have network field in CR"
+// +kubebuilder:validation:XValidation:rule="!has(self.external) || (has(self.external) && ((!has(self.external.enable) || self.external.enable == false )|| (self.external.enable == true && (!has(self.storage)))))",message="external mode enabled cluster cannot have storage field in CR"
```

Comment:
Is there any way to simplify the statements when there are nullable fields? It would be nice if it just treated values of nullable fields as false or 0 if not set. This is a lot of complexity with nullable fields.

Reply:
No, I don't we can make it simpler `field presence can be checked via has(self.field)`. We have `   // +kubebuilder:validation:Optional` but will be valid for only one settings but not when we compare with multiple settings